src/lib/core/run.js

```diff
@@ -136,7 +138,7 @@ function run(refs, schema, container) {
 
       if (typeof sub.$ref === 'string') {
         // increasing depth only for repeated refs seems to be fixing #258
-        if (sub.$ref === '#' || (lastRef === sub.$ref && ++depth > random.number(0, 3))) {
+        if (sub.$ref === '#' || (lastRef === sub.$ref && ++depth > random.number(refDepthMin, refDepthMax) - 1)) {
```

Comment:
I can remove the `-1` here if desired -- it depends on how one defines "depth" here -- whether the maximum depth should be empty, or whether it should be fully rendered, with one additional, one-level-deeper, empty element in it.

With `-1`, with `refDepthMin` and `refDepthMax` both set to 4:

```
{
  "collection": {
    "a": {
      "subA": [
        {
          "subA": [
            {
              "subA": [
                {
                  "subA": []
                }
              ]
            }
          ]
        }
      ]
    }
  }
}
```

Without `-1`, with `refDepthMin` and `refDepthMax` both set to 4:

```
{
  "collection": {
    "a": {
      "subA": [
        {
          "subA": [
            {
              "subA": [
                {
                  "subA": [
                    {
                      "subA": []
                    }
                  ]
                }
              ]
            }
          ]
        }
      ]
    }
  }
}
```

Reply:
The first of those looks preferable to me, of those two, so I included `-1` in the PR.

Reply:
Yeah, I liked, but can you do me a favor?

Please validate both values, ensure `refDepthMin` is always less-or-equal (?) than `refDepthMax` and then, just subtract the -1 as desired, e.g.

```js
const maxDepth = random.number(refDepthMin, refDepthMax) - 1;
// ...
if ( ... ++depth > maxDepth)
// ...
```

Reply:
Sure! I've pushed several commits, ending with d9537331a, just now. Do those changes address what you're looking for?